src/message/ersip_hdr_event.erl

```diff
@@ -41,6 +43,17 @@
 %% API
 %%===================================================================
 
+-spec new(binary()) -> event().
```

Comment:
new is consturction from internal data structures. So

```
-spec new(type()) -> event().
-spec new(type(), id()) -> event().
```
is agree with this approach

To use binary - pleas use make functions. For example:
```
%% Create full header value from binary
-spec make(binary()) -> event().
%% Create ID from binary:
-spec make_id(binary()) -> id().
-spec make_type(binary()) -> type().
````

Reply:
ok

Reply:
renamed to `make_with_type`